# HR Analytics: Workforce Performance & Employee Insights

## 01 - Data Understanding

### Project Overview

This project analyzes employee workforce data to uncover insights related to employee performance, compensation, engagement, absenteeism, recruitment, and termination patterns.

The goal is to support HR decision-making through data analysis, SQL business questions, and an executive Power BI dashboard.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 100)
pd.set_option("display.float_format", "{:,.2f}".format)

In [2]:
df = pd.read_csv("../data/HRDataset_v14.csv")

df.head()

,Employee_Name,EmpID,MarriedID,MaritalStatusID,GenderID,EmpStatusID,DeptID,PerfScoreID,FromDiversityJobFairID,Salary,Termd,PositionID,Position,State,Zip,DOB,Sex,MaritalDesc,CitizenDesc,HispanicLatino,RaceDesc,DateofHire,DateofTermination,TermReason,EmploymentStatus,Department,ManagerName,ManagerID,RecruitmentSource,PerformanceScore,EngagementSurvey,EmpSatisfaction,SpecialProjectsCount,LastPerformanceReview_Date,DaysLateLast30,Absences
0,"Adinolfi, Wilson K",10026,0,0,1,1,5,4,0,62506,0,19,Production Technician I,MA,1960,07/10/83,M,Single,US Citizen,No,White,7/5/2011,NaN,N/A-StillEmployed,Active,Production,Michael Albert,22.00,LinkedIn,Exceeds,4.60,5,0,1/17/2019,0,1
1,"Ait Sidi, Karthikeyan",10084,1,1,1,5,3,3,0,104437,1,27,Sr. DBA,MA,2148,05/05/75,M,Married,US Citizen,No,White,3/30/2015,6/16/2016,career change,Voluntarily Terminated,IT/IS,Simon Roup,4.00,Indeed,Fully Meets,4.96,3,6,2/24/2016,0,17
2,"Akinkuolie, Sarah",10196,1,1,0,5,5,3,0,64955,1,20,Production Technician II,MA,1810,09/19/88,F,Married,US Citizen,No,White,7/5/2011,9/24/2012,hours,Voluntarily Terminated,Production,Kissy Sullivan,20.00,LinkedIn,Fully Meets,3.02,3,0,5/15/2012,0,3
3,"Alagbe,Trina",10088,1,1,0,1,5,3,0,64991,0,19,Production Technician I,MA,1886,09/27/88,F,Married,US Citizen,No,White,1/7/2008,NaN,N/A-StillEmployed,Active,Production,Elijiah Gray,16.00,Indeed,Fully Meets,4.84,5,0,1/3/2019,0,15
4,"Anderson, Carol",10069,0,2,0,5,5,3,0,50825,1,19,Production Technician I,MA,2169,09/08/89,F,Divorced,US Citizen,No,White,7/11/2011,9/6/2016,return to school,Voluntarily Terminated,Production,Webster Butler,39.00,Google Search,Fully Meets,5.00,4,0,2/1/2016,0,2


In [3]:
df.shape

(311, 36)

#### Dataset Dimensions

The first step is to understand the size of the dataset by identifying the number of rows (observations) and columns (features).

In [4]:
print(f"Number of Rows: {df.shape[0]}")
print(f"Number of Columns: {df.shape[1]}")

Number of Rows: 311
Number of Columns: 36


#### Dataset Columns

Display all available variables contained in the dataset.

In [5]:
df.columns.tolist()

['Employee_Name',
 'EmpID',
 'MarriedID',
 'MaritalStatusID',
 'GenderID',
 'EmpStatusID',
 'DeptID',
 'PerfScoreID',
 'FromDiversityJobFairID',
 'Salary',
 'Termd',
 'PositionID',
 'Position',
 'State',
 'Zip',
 'DOB',
 'Sex',
 'MaritalDesc',
 'CitizenDesc',
 'HispanicLatino',
 'RaceDesc',
 'DateofHire',
 'DateofTermination',
 'TermReason',
 'EmploymentStatus',
 'Department',
 'ManagerName',
 'ManagerID',
 'RecruitmentSource',
 'PerformanceScore',
 'EngagementSurvey',
 'EmpSatisfaction',
 'SpecialProjectsCount',
 'LastPerformanceReview_Date',
 'DaysLateLast30',
 'Absences']

#### Data Types

Understanding the data type of each variable helps identify numerical, categorical, and date-related features.

In [6]:
df.dtypes

Employee_Name                  object
EmpID                           int64
MarriedID                       int64
MaritalStatusID                 int64
GenderID                        int64
EmpStatusID                     int64
DeptID                          int64
PerfScoreID                     int64
FromDiversityJobFairID          int64
Salary                          int64
Termd                           int64
PositionID                      int64
Position                       object
State                          object
Zip                             int64
DOB                            object
Sex                            object
MaritalDesc                    object
CitizenDesc                    object
HispanicLatino                 object
RaceDesc                       object
DateofHire                     object
DateofTermination              object
TermReason                     object
EmploymentStatus               object
Department                     object
ManagerName 

#### Dataset Information

The dataset information provides the number of non-null values, memory usage, and data types for every feature.

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 311 entries, 0 to 310
Data columns (total 36 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Employee_Name               311 non-null    object 
 1   EmpID                       311 non-null    int64  
 2   MarriedID                   311 non-null    int64  
 3   MaritalStatusID             311 non-null    int64  
 4   GenderID                    311 non-null    int64  
 5   EmpStatusID                 311 non-null    int64  
 6   DeptID                      311 non-null    int64  
 7   PerfScoreID                 311 non-null    int64  
 8   FromDiversityJobFairID      311 non-null    int64  
 9   Salary                      311 non-null    int64  
 10  Termd                       311 non-null    int64  
 11  PositionID                  311 non-null    int64  
 12  Position                    311 non-null    object 
 13  State                       311 non

#### Statistical Summary

Generate descriptive statistics for all numerical variables.

In [8]:
df.describe()

,EmpID,MarriedID,MaritalStatusID,GenderID,EmpStatusID,DeptID,PerfScoreID,FromDiversityJobFairID,Salary,Termd,PositionID,Zip,ManagerID,EngagementSurvey,EmpSatisfaction,SpecialProjectsCount,DaysLateLast30,Absences
count,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,303.00,311.00,311.00,311.00,311.00,311.00
mean,"10,156.00",0.40,0.81,0.43,2.39,4.61,2.98,0.09,"69,020.68",0.33,16.85,"6,555.48",14.57,4.11,3.89,1.22,0.41,10.24
std,89.92,0.49,0.94,0.50,1.79,1.08,0.59,0.29,"25,156.64",0.47,6.22,"16,908.40",8.08,0.79,0.91,2.35,1.29,5.85
min,"10,001.00",0.00,0.00,0.00,1.00,1.00,1.00,0.00,"45,046.00",0.00,1.00,"1,013.00",1.00,1.12,1.00,0.00,0.00,1.00
25%,"10,078.50",0.00,0.00,0.00,1.00,5.00,3.00,0.00,"55,501.50",0.00,18.00,"1,901.50",10.00,3.69,3.00,0.00,0.00,5.00
50%,"10,156.00",0.00,1.00,0.00,1.00,5.00,3.00,0.00,"62,810.00",0.00,19.00,"2,132.00",15.00,4.28,4.00,0.00,0.00,10.00
75%,"10,233.50",1.00,1.00,1.00,5.00,5.00,3.00,0.00,"72,036.00",1.00,20.00,"2,355.00",19.00,4.70,5.00,0.00,0.00,15.00
max,"10,311.00",1.00,4.00,1.00,5.00,6.00,4.00,1.00,"250,000.00",1.00,30.00,"98,052.00",39.00,5.00,5.00,8.00,6.00,20.00


#### Categorical Variables Summary

Display descriptive statistics for categorical features.

In [10]:
df.describe(include="object")

,Employee_Name,Position,State,DOB,Sex,MaritalDesc,CitizenDesc,HispanicLatino,RaceDesc,DateofHire,DateofTermination,TermReason,EmploymentStatus,Department,ManagerName,RecruitmentSource,PerformanceScore,LastPerformanceReview_Date
count,311,311,311,311,311,311,311,311,311,311,104,311,311,311,311,311,311,311
unique,311,32,28,307,2,5,3,4,6,101,96,18,3,6,21,9,4,137
top,"Adinolfi, Wilson K",Production Technician I,MA,09/09/65,F,Single,US Citizen,No,White,1/10/2011,8/19/2018,N/A-StillEmployed,Active,Production,Michael Albert,Indeed,Fully Meets,1/14/2019
freq,1,137,276,2,176,137,295,282,187,14,2,207,207,209,22,87,243,18


#### Random Sample

Display a random sample of employees to better understand the structure of the dataset.

In [11]:
df.sample(5, random_state=42)

,Employee_Name,EmpID,MarriedID,MaritalStatusID,GenderID,EmpStatusID,DeptID,PerfScoreID,FromDiversityJobFairID,Salary,Termd,PositionID,Position,State,Zip,DOB,Sex,MaritalDesc,CitizenDesc,HispanicLatino,RaceDesc,DateofHire,DateofTermination,TermReason,EmploymentStatus,Department,ManagerName,ManagerID,RecruitmentSource,PerformanceScore,EngagementSurvey,EmpSatisfaction,SpecialProjectsCount,LastPerformanceReview_Date,DaysLateLast30,Absences
290,"Vega, Vincent",10144,0,2,1,1,5,3,0,88976,0,17,Production Manager,MA,2169,10/10/68,M,Divorced,US Citizen,No,White,8/1/2011,NaN,N/A-StillEmployed,Active,Production,Janet King,2.00,Employee Referral,Fully Meets,3.93,3,0,2/27/2019,0,19
9,"Bacong, Alejandro",10250,0,2,1,1,3,3,0,50178,0,14,IT Support,MA,1886,01/07/88,M,Divorced,US Citizen,No,White,1/5/2015,NaN,N/A-StillEmployed,Active,IT/IS,Peter Monroe,7.00,Indeed,Fully Meets,5.00,5,6,2/18/2019,0,16
57,"Costello, Frank",10193,1,1,1,1,3,3,0,83552,0,9,Data Analyst,MA,1810,08/26/86,M,Married,US Citizen,No,White,3/30/2015,NaN,N/A-StillEmployed,Active,IT/IS,Simon Roup,4.00,Indeed,Fully Meets,3.04,3,6,1/22/2019,0,2
60,"Daneault, Lynn",10099,0,0,0,1,6,3,0,65729,0,21,Sales Manager,VT,5473,04/19/90,F,Single,US Citizen,No,White,5/5/2014,NaN,N/A-StillEmployed,Active,Sales,Debra Houlihan,15.00,Indeed,Fully Meets,4.62,4,0,1/24/2019,0,8
25,"Booth, Frank",10199,0,0,1,4,3,3,0,103613,1,30,Enterprise Architect,CT,6033,07/30/64,M,Single,US Citizen,No,Black or African American,2/17/2014,2/19/2016,Learned that he is a gangster,Terminated for Cause,IT/IS,Simon Roup,4.00,LinkedIn,Fully Meets,3.50,5,7,1/10/2016,0,2


#### Initial Observations

The dataset contains 311 employee records and 36 variables covering employee demographics, employment status, compensation, department, position, recruitment source, performance, engagement, satisfaction, absences, and termination information.

The dataset supports several business analysis areas:

- Workforce demographics
- Employee attrition and termination analysis
- Department and position analysis
- Salary and compensation analysis
- Recruitment source effectiveness
- Employee performance and satisfaction
- Absenteeism and lateness patterns

#### Notebook Summary

This notebook introduced the dataset and reviewed its structure, variables, data types, and initial statistical summaries.

The next notebook will focus on data quality assessment, including missing values, duplicate records, date formatting, and consistency checks.